examples/step-5/step-5.cc

```diff
@@ -569,7 +402,43 @@ void Step5<dim>::run ()
   // two-dimensional triangulation, while this function is a template for
   // arbitrary dimension. Since this is only a demonstration program, we will
   // not use different input files for the different dimensions, but rather
-  // kill the whole program if we are not in 2D:
+  // kill the whole program if we are not in 2D. Of course, since the main
+  // function below assumes that we are working in two dimensions we could
+  // skip this check, in this version of the program, without any ill effects.
+  //
+  // It turns out that more than 90 per cent of programming errors are invalid
+  // function parameters such as invalid array sizes, etc, so we use
+  // assertions heavily through out deal.II to catch such mistakes. For this,
+  // the <code>Assert</code> macro is a good choice, since it makes sure that
+  // the condition which is given as first argument is valid, and if not
+  // throws an exception (its second argument) which will usually terminate
+  // the program giving information where the error occurred and what the
+  // reason was. This generally reduces the time to find programming errors
+  // dramatically and we have found assertions an invaluable means to program
+  // fast.
+  //
+  // On the other hand, all these checks (there are over 9000 of them in the
+  // library at present) should not slow down the program too much if you want
+  // to do large computations. To this end, the <code>Assert</code> macro is
+  // only used in debug mode and expands to nothing if in optimized
+  // mode. Therefore, while you test your program on small problems and debug
+  // it, the assertions will tell you where the problems are. Once your
+  // program is stable, you can switch off debugging and the program will run
+  // your real computations without the assertions and at maximum speed. (In
+  // fact, it turns out the switching off all the checks in the library that
```

Comment:
that switching off?

The sentence is long and convoluted. Feel free to rephrase.